src/irr/core/CDoublyLinkedList.h

```diff
@@ -0,0 +1,83 @@
+#ifndef __DOUBLY_LINKED_LIST_H_INCLUDED__
+#define __DOUBLY_LINKED_LIST_H_INCLUDED__
+#include "irr/core/alloc/PoolAddressAllocator.h"
+#include "irr/core/Types.h"
+
+namespace irr {
+	namespace core {
+
+		template<typename Value>
+		class DoublyLinkedList
+		{
+			struct Snode
+			{
+				struct Snode* prev;
+				Value data;
+				struct Snode* next;
+
+				Snode(Value v) { data = v; }
+			};
+
+			uint32_t cap;
+			Snode* begin;
+			Snode* end;
+			PoolAddressAllocator<uint32_t> alloc;
+
+			inline void popBack()
+			{
+				if(end->prev != nullptr)
+					end->prev->next = nullptr;
+				Snode* temp = end;
+				end = end->prev;
+				alloc.free_addr(reinterpret_cast<uint32_t>(temp), sizeof(Snode));
```

Comment:
you need to explictly call the destructor on the node you're removing, `Value` may be a non-trivial type